### PR TITLE
Introduce method to concat list items

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Add a method to concat list items.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/structs/List.js [hunk 1]
```diff
@@ -53,6 +53,30 @@ module.exports = class List {
...
+    // Prepend current list
+    items.unshift(this);
+
+    // Get items from lists
+    items = items.map(function (item) {
+      if ((item instanceof List)) {
+        return item.getItems();
+      }
+
+      return item
```
Comment (aldipower) [inline src/js/structs/List.js]: :(

File: src/js/structs/List.js [hunk 2]
```diff
@@ -53,6 +53,30 @@ module.exports = class List {
...
+   * @returns {List} List (or child class) containing concatenated items
+   */
+  concatItems(...items) {
+
+    // Prepend current list
+    items.unshift(this);
+
+    // Get items from lists
+    items = items.map(function (item) {
+      if ((item instanceof List)) {
```
Comment (aldipower) [inline src/js/structs/List.js]: Double parenthesis.
Looks pretty nice actually. ;)

File: src/js/structs/List.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/structs/List.js]: I don't do this to offend you, but there is no eslint rule to enforce the semicolon and my IDE is not adding them as they aren't required by the ECMAScript (Automatic Semicolon Insertion).

**Comment by aldipower** [comment]
I think this is the way to do it, so I am fine with it.

---
Not your concern, but I _don't_ like to introduce something probably very non-performant for a simple problem that is already solved very good.
Just want to shout this into the room.

File: src/js/structs/List.js (hunk 1 above)
Comment (aldipower) [inline src/js/structs/List.js]: I know. But to be honest, you are the only one, where I notice this regularly, so it is a little bit annoying. Nevermind.

**Comment by rcorral** [comment]
@orlandohohmeier what's the specific use case?

File: src/js/structs/List.js [hunk 3]
```diff
@@ -53,6 +53,30 @@ module.exports = class List {
...
+   */
+  concatItems(...items) {
+
+    // Prepend current list
+    items.unshift(this);
+
+    // Get items from lists
+    items = items.map(function (item) {
+      if (item instanceof List) {
+        return item.getItems();
```
Comment (pierlo-upitup) [inline src/js/structs/List.js]: Don't we have a way to get a flattened array of `items` already from an arbitrarily nested list?

Comment (orlandohohmeier) [inline src/js/structs/List.js]: Not, that I'm aware of...

**Comment by orlandohohmeier** [comment]
@rcorral I needed something to concat three different lists (to build the job history) and was looking for something nicer than the following:

```
new List({items:[].concat(FirstList.getItems(),SecondList.getItems(),ThirdList.getItems()})
```

But I'll probably merge the items in a different place (before creating lists from them) and don't need this method any more.

**Comment by kennyt** [comment]
so what's the deal with this PR now? is it unnecessary?

**Comment by orlandohohmeier** [comment]
Yep, it's unnecessary. Closing it.

**Pull request closed without merging**